src/MongoDB.Bson/BsonExtensionMethods.cs

```diff
@@ -43,6 +43,7 @@ public static class BsonExtensionMethods
             BsonSerializationArgs args = default(BsonSerializationArgs)
             )
         {
+            args.SetOrValidateNominalType(typeof(TNominalType), "<TNominalType>");
```

Comment:
Sorry, I've added commetns several minutes before and didn't notice that you've added a new commit.
However, my comment: https://github.com/rstam/mongo-csharp-driver/pull/136#discussion_r358845401 still makes sense to me. In case if we have a single error message, we won't need to have both validations with the same target. For example in lines 46 and 74

Reply:
I would rather the error messages be slightly different and more explanatory.

We can still have code sharing and customize the error messages as in the new `SetOrValidateNominalType` method.

Reply:
I agree that code sharing is a good idea, and don't really agree about the same messages and as a result double validation in some cases. But I also think that there is no harm to have only the currently added solution, so I'm ok.